torchtitan/metrics.py

```diff
@@ -16,6 +16,13 @@
 from torchtitan.parallelisms import ParallelDims
 from torchtitan.utils import device_module, device_type
 
+# Optional wandb import
+try:
+    import wandb
+
+    WANDB_AVAILABLE = True
+except ImportError:
+    WANDB_AVAILABLE = False
```

Comment:
let's raise the exception similar to https://github.com/pytorch/torchtitan/blob/main/torchtitan/float8.py#L46
to avoid silently missing the logs

Reply:
Went for something a bit different here

[rank0]:2024-12-02 16:16:43,791 - root - ERROR - Failed to create WandB logger: No module named 'wandb'. Please install it using 'pip install wandb'.